### PR TITLE
feat: add integration with react-hook-form

### PR DESCRIPTION
**Pull request by spawnia**

### CODE REVIEW

File: src/Fields/FieldWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+export function FieldWrapper<
+  TFieldValues extends FieldValues,
+  TName extends FieldPath<TFieldValues>,
+>(props: FieldWrapperProps<TFieldValues, TName>) {
+  const { fieldState } = useController<TFieldValues, TName>(props.controller);
+
+  return (
+    <Form.Item
+      // hasFeedback={fieldState.invalid} TODO uncomment when antd3 has been removed from limes-frontend
```
Comment (SimBig) [inline src/Fields/FieldWrapper.tsx:32]: can we solve this somehow via the FormProvider? In NeMo we have no dependency on antd3.

**Review by SimBig — CHANGES_REQUESTED** [review]

File: src/Fields/FieldWrapper.tsx (hunk 1 above)
Comment (mic-web) [reply src/Fields/FieldWrapper.tsx]: Yes good idea to make it configurable with the `FieldProvider`.  Context could be extended to:


```
{
  disabled: boolean;
  formItemProps:  (fieldState: ControllerFieldState) => Partial<FormItemProps>;
}
```

Comment (spawnia) [reply src/Fields/FieldWrapper.tsx]: @mic-web you can just push to this pull request.

Comment (mic-web) [reply src/Fields/FieldWrapper.tsx]: Alternative to `Partial` might be possible

Comment (mic-web) [reply src/Fields/FieldWrapper.tsx]: Done

File: src/Fields/FieldProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  disabled,
+  formItemProps,
+}: FieldProviderProps) {
+  // When current provider does not specify optional props, the parent field
+  // provider's values will be used
+  const parentFieldContext = useFieldContext();
+
+  const value = useMemo(
+    () => ({
+      disabled: (disabled || parentFieldContext.disabled) ?? false,
```
Comment (spawnia) [inline src/Fields/FieldProvider.tsx:45]:
```suggestion
      disabled: disabled ?? parentFieldContext.disabled ?? false,
```

Or else `disabled = false` would be overruled by `parentFieldContext.disabled`

File: src/Fields/FieldProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      formItemProps:
+        (formItemProps || parentFieldContext.formItemProps) ??
+        formItemFieldProps,
```
Comment (spawnia) [inline src/Fields/FieldProvider.tsx:48]:
```suggestion
        formItemProps ?? parentFieldContext.formItemProps ?? formItemFieldProps,
```

File: src/Fields/TextAreaField.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import { TextAreaProps } from 'antd/lib/input/TextArea';
```
Comment (spawnia) [inline src/Fields/TextAreaField.tsx:1]: We should expose that from `../Input`

File: src/Fields/index.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+// Not exported from src/index.ts in order to keep react-hook-form optional
```
Comment (spawnia) [inline src/Fields/index.ts:1]:
```suggestion
```

**Comment by mic-web** [comment]
https://github.com/mll-lab/react-components/pull/176/commits/86b1ff2d189aff831ea1f0a63b213047180fe1ec

Can we remove it like committed? In my opinion it doesn't help when using Typescript, but leads to propblems with props definitions in stories

**Comment by spawnia** [comment]
Let's change eslint in https://github.com/mll-lab/eslint-config - please add a pull request there.

**Pull request merged**